core/runtime/src/main/java/io/quarkus/runtime/graal/WindowsDiagnosticPrinter.java

```diff
@@ -0,0 +1,37 @@
+package io.quarkus.runtime.graal;
+
+import static org.graalvm.nativeimage.Platform.HOSTED_ONLY;
+import static org.graalvm.nativeimage.Platform.WINDOWS;
+
+import java.io.PrintStream;
+
+import org.graalvm.nativeimage.ImageSingletons;
+import org.graalvm.nativeimage.Platforms;
+import org.graalvm.nativeimage.hosted.Feature;
+
+import com.oracle.svm.core.annotate.AutomaticFeature;
+
+/**
+ * A signal handler that prints diagnostic thread info to standard output.
+ */
+@Platforms(WINDOWS.class)
+public final class WindowsDiagnosticPrinter extends DiagnosticPrinter {
+
+    @Platforms(HOSTED_ONLY.class)
+    WindowsDiagnosticPrinter() {
+    }
+
+    @Override
+    protected void printTid(PrintStream w, Thread thread) {
+        w.println(" tid=(unknown)");
```

Comment:
I'm not satisfied with this, but I didn't find a way in [WindowsJavaThreads](https://github.com/oracle/graal/blob/master/substratevm/src/com.oracle.svm.core.windows/src/com/oracle/svm/core/windows/WindowsJavaThreads.java) to retrieve the Windows native thread ID. Suggestions are welcome here.

Reply:
Look at this gist from https://gist.github.com/rednaxelafx/843622 to access the Native thread ID: If `sun.jvm.hotspot.tools.Tool` is available in Quarkus you can get it with `ThreadInfo` for any platform:

```
        List<ThreadInfo> infos = ThreadUtils.getAllThreadInfos();
        for (ThreadInfo info : infos) {
            System.out.printf("Thread@%s: tid=%d nid=0x%x\n",
            	info.getAddress(), info.getTid(), info.getNid());
        }
```

You can extend `Tool` within an inner static class in  `WindowsJavaThreads`..